src/js/actions/layers.js

```diff
@@ -1377,45 +1382,147 @@ define(function (require, exports) {
     };
 
     /**
+     * Helper function to find the right most artboard 
+     * @param {Immutable.List.of<Layer>} artboards
+     * @return {Layer}
+     */
+
+    var _getRightMost = function (artboards) {
```

Comment:
I suggest naming this instead `_findRightMostArtboard`. Putting `Artboard` in the name makes it obvious what it's finding (even without reading the docs), and `find` hints that this is an O(N) search instead of a constant-time lookup.